src/cryptonote_core/service_node_quorum_cop.cpp

```diff
@@ -424,12 +425,25 @@ namespace service_nodes
       return true;
     }
 
-    vvc = {};
-    std::shared_ptr<const testing_quorum> quorum = m_core.get_testing_quorum(vote.type, vote.block_height);
+    uint64_t quorum_height = vote.block_height;
+    if (vote.type == quorum_type::checkpointing)
+    {
+      if (vote.block_height < REORG_SAFETY_BUFFER_BLOCKS_POST_HF12)
+      {
+        vvc.m_invalid_block_height = true;
+        LOG_ERROR("Quorum state for height: " << vote.block_height << " was not cached in daemon!");
```

Comment:
This error message looks odd given the `if` guarding it -- perhaps it should just be `"Invalid vote height: " << ...`